base/libgit2/oid.jl

```diff
@@ -139,6 +139,8 @@ function GitHash(obj::GitObject)
     GitHash(ccall((:git_object_id, :libgit2), Ptr{UInt8}, (Ptr{Cvoid},), obj.ptr))
 end
 
+==(obj1::GitObject, obj2::GitObject) = GitHash(obj1) == GitHash(obj2)
```

Comment:
This seemed like a reasonable thing to do, but is not really required for this PR (it simplifies a test).

Reply:
It seems reasonable to me as well, so long as there's a test for it. :slightly_smiling_face: